src/index.ts

```diff
@@ -71,12 +71,17 @@ function processGetAddrResponse(response: Buffer) {
   let hash: Buffer | undefined
   if(PKLEN != ED25519_PK_SIZE) {
     hash = Buffer.from(partialResponse.slice(0, 20))
-  }
 
-  //"advance" buffer
-  partialResponse = partialResponse.slice(20)
+    //"advance" buffer
+    partialResponse = partialResponse.slice(20)
+  }
 
-  const address = Buffer.from(partialResponse.subarray(0, -2)).toString()
+  let address: Buffer | string
+  if(PKLEN == ED25519_PK_SIZE) {
+    address = Buffer.from(partialResponse.subarray(0, -2))
+  } else {
+    address = Buffer.from(partialResponse.subarray(0, -2)).toString()
+  }
```

Comment:
Is this what we expect to do? String or Buffer, based on the address type? shouldn't we go with string or buffer to be consistent?

Reply:
@emmanuelm41  Was waitting for your opºinion on this to be honest.
For ed25519 we have an address of type "004553453453" for secp256k1 "Pavax-fkekeo00ei2", so one is an hex array the other a string.

Reply:
I would go both with strings. ed25519 would be hex string, and secp256k1 would be just plain string. I think it will be more consistent. wdyt?

Reply:
I agree, also easier to use the response from the method